specification/elasticsan/resource-manager/Microsoft.ElasticSan/stable/2023-01-01/elasticsan.json

```diff
@@ -2205,6 +2662,98 @@
         }
       },
       "description": "List of private endpoint connections associated with SAN"
+    },
+    "Snapshot": {
+      "type": "object",
+      "required": [
+        "properties"
+      ],
+      "description": "Response for Volume Snapshot request.",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Properties of Volume Snapshot.",
+          "$ref": "#/definitions/SnapshotProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
+        }
+      ]
+    },
+    "SnapshotList": {
+      "type": "object",
+      "description": "List of Snapshots",
+      "additionalProperties": false,
+      "properties": {
+        "value": {
+          "description": "An array of Snapshot objects.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Snapshot"
+          }
+        },
+        "nextLink": {
+          "description": "URI to fetch the next section of the paginated response.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "SnapshotProperties": {
+      "type": "object",
+      "required": [
+        "creationData"
+      ],
+      "description": "Properties for Snapshot.",
+      "properties": {
+        "creationData": {
+          "description": "Data used when creating a volume snapshot.",
+          "$ref": "#/definitions/SnapshotCreationData"
+        },
+        "provisioningState": {
```

Comment:
Is this required on the request body?

Reply:
As we have same request & response model, it is needed.
But yeah it hasn't been marked readOnly, will update it. Thanks